src/components/BMDashboard/Login/BMLogin.jsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { Redirect, withRouter } from 'react-router-dom';
+import { Form, FormGroup, FormText, Input, Label, Button, FormFeedback } from 'reactstrap';
+import Joi from 'joi';
+
+import { loginBMUser } from 'actions/authActions';
+
+const BMLogin = (props) => {
```

Comment:
You are already destructuring props at the beginning of your component. However, you can further improve it by destructuring individual props directly in the function parameters:
const BMLogin = ({ dispatch, auth, history, location }) => {
  // ...
}